crypto/ledger_secp256k1.go

```diff
@@ -214,3 +249,27 @@ func getPubKey(device LedgerSECP256K1, path hd.BIP44Params) (tmcrypto.PubKey, er
 
 	return compressedPublicKey, nil
 }
+
+// getPubKeyAddr reads the pubkey and the address from a ledger device
+// This function is marked as Safe as it will require user confirmation and
+// account and index will be shown in the device
+//
+// since this involves IO, it may return an error, which is not exposed
+// in the PubKey interface, so this function allows better error handling
+func getPubKeyAddrSafe(device LedgerSECP256K1, path hd.BIP44Params, hrp string) (tmcrypto.PubKey, string, error) {
+	publicKey, addr, err := device.GetAddressPubKeySECP256K1(path.DerivationPath(), hrp)
+	if err != nil {
+		return nil, "", fmt.Errorf("please open Cosmos app on the Ledger device - error: %v", err)
```

Comment:
There renders the following if the address is rejected:
```
ERROR: please open Cosmos app on the Ledger device - error: [APDU_CODE_COMMAND_NOT_ALLOWED] Command not allowed (no current EF)
```
```suggestion
		return nil, "", fmt.Errorf("address %s rejected", addr)
```